app.js

```diff
@@ -4,38 +4,107 @@ const radioButtons = document.querySelectorAll('input');
 const nextButton = document.querySelector('button');
 const caughtSpan = document.querySelector('#caught-span');
 const images = document.querySelectorAll('label > img');
+const nextDiv = document.querySelector('#next');
 
-// initialize state
-let pokemonCaught = 0;
+export let caughtPokemon = [];
+export let pokemonCaught = 0;
+
+
+function findById(someArray, someId) {
+    for (const booger of someArray){
+        if(booger.id === someId){
+            return booger;
+        }}}
 
 function getRandomPokemon(array){
     const index = Math.floor(Math.random() * array.length);
     
     return array[index];
 }
-let pokemon1 = getRandomPokemon(rawPokeData)
-let pokemon2 = getRandomPokemon(rawPokeData)
-let pokemon3 = getRandomPokemon(rawPokeData)
 
-while (pokemon1.id === pokemon2.id) {
-    pokemon1 = getRandomPokemon(rawPokeData);
-}
 
-while (pokemon2.id === pokemon3.id) {
-    pokemon2 = getRandomPokemon(rawPokeData);
-}
-while (pokemon3.id === pokemon1.id) {
-    pokemon3 = getRandomPokemon(rawPokeData);
+
+function setupGame() {
+    if (pokemonCaught === 10) {
+        const stringyMon = JSON.stringify(caughtPokemon);
+        localStorage.setItem('POKEMON', stringyMon);
+        // window.location.href = './results-page/index.html';
+    }
+    
+    nextDiv.classList.add('hidden');
+
+    for (let i = 0; i < radioButtons.length; i++) {
+        radioButtons[i].disabled = false;
+        radioButtons[i].checked = false;
+        images[i].style.opacity = 1;
+    }
+
+
+    
+
+    let pokemon1 = getRandomPokemon(rawPokeData)
+    let pokemon2 = getRandomPokemon(rawPokeData)
+    let pokemon3 = getRandomPokemon(rawPokeData)
+
+
+
+    while (pokemon1.id === pokemon2.id) {
```

Comment:
You can put all the while logic in one area. Example:
while(pokemon1.id === pokemon2.id || pokemon2.id === pokemon3.id || pokemon3.id === pokemon1.id) { 
          pokemon2 = getRandomPokemon(rawPokeData);
          pokemon3 = getRandomPokemon(rawPokeData);
}
You had a duplicate and I believe this might solve that problem. (you need one Pokemon to stay consistent to compare to. (pokemon1 in this case)

OOOOOORRRRRrrrr.....

    while (pokemon1.id === pokemon2.id) {
        pokemon2 = getRandomPokemon(rawPokeData);
    }
    while (pokemon2.id === pokemon3.id || pokemon3.id === pokemon1.id) {
        pokemon3 = getRandomPokemon(rawPokeData);
    }